### PR TITLE
fix: Open OO file on other instance in shared drive

### PR DESCRIPTION
**Pull request by zatteo**
Due to the org B2B instance name ending like user B2B instance name, it was opened in same instance.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved detection of whether documents should open in another instance.
  * Correctly handles missing instance URLs and compares hostnames from document and current instance data (including cases without explicit ports and organization-based hosts).
  * Avoids unnecessary redirects when instance hosts match.

* **Tests**
  * Extended test coverage for matching and differing instance hostnames across supported URL formats, including edge-case URL formatting.

### CODE REVIEW

File: src/modules/views/OnlyOffice/helpers.js [hunk 1]
```diff
@@ -133,7 +133,10 @@ export const makeOnlyOfficeFileRoute = (
  * @returns {boolean}
  */
 export const shouldBeOpenedOnOtherInstance = ({ data }, instanceUri) => {
-  return !!instanceUri && !instanceUri.includes(data.attributes.instance)
+  if (!instanceUri) return false
+  const docHost = data.attributes.instance.split(':')[0]
+  const currentHost = new URL(instanceUri).hostname
+  return docHost !== currentHost
```
[the remark this answers is not in the document]
Comment (zatteo) [reply src/modules/views/OnlyOffice/helpers.js:139]: The issue is that data.attributes.instance looks like `myorgc81729.example.com:8080` so it is not parsed correctly by new URL

**Review by shepilov — APPROVED** [review]
